src/MSBuild.UnitTests/XMake_Tests.cs

```diff
@@ -523,6 +523,87 @@ public void Help(string indicator)
             .ShouldBe(MSBuildApp.ExitType.Success);
         }
 
+        [Fact]
+        public void VersionSwitch()
+        {
+            List<string> cmdLine = new()
+            {
+#if !FEATURE_RUN_EXE_IN_TESTS
+                EnvironmentProvider.GetDotnetExePath(),
+#endif
+                FileUtilities.EnsureDoubleQuotes(RunnerUtilities.PathToCurrentlyRunningMsBuildExe),
+                "-nologo",
+                "-version"
+            };
+
+            using Process process = new()
+            {
+                StartInfo =
+                {
+                    FileName = cmdLine[0],
+                    Arguments = string.Join(" ", cmdLine.Skip(1)),
+                    UseShellExecute = false,
+                    RedirectStandardOutput = true,
+                },
+            };
+
+            process.Start();
+            process.WaitForExit();
+            process.ExitCode.ShouldBe(0);
+
+            string output = process.StandardOutput.ReadToEnd();
+            // Change Version switch output to finish with a newline https://github.com/dotnet/msbuild/pull/9485
+            if (ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_10))
```

Comment:
May I clarify if there is a case for this test case, when the feature will not be available => hence the output will not contain the new line? Since the feature is opt-out, new line will be written to output unless change wave is configured, which is tested in separate test: VersionSwitchDisableChangeWave.

Reply:
Thank you for the explanation, basically the test covers both scenarios relying on the setup of local environment.
I would suggest not to rely on the local environment in unit tests, and cover in this test success path of the ShowVersion method when there is no ChangeWaves enabled. As you suggested to remove env var if exists and proceed to the test :)

Reply:
Our `TestEnvironment` class should clear the changewave state when it starts up--but today it does so only when it _cleans_ up, which is awkward and doesn't provide a strong guarantee.

Reply:
Committed changes to use TestEnvironment. PR builds passed.